openlibrary/solr/update_work.py

```diff
@@ -1481,13 +1484,23 @@ def _solr_update(requests: List[SolrUpdateRequest], commitWithin=60000):
     # update works
     requests: List[SolrUpdateRequest] = []
     requests += [DeleteRequest(deletes)]
-    for k in wkeys:
-        logger.debug("updating work %s", k)
+
+    def handle_wkey(wkey, requests):
+        logger.debug("updating work %s", wkey)
         try:
-            w = data_provider.get_document(k)
+            w = data_provider.get_document(wkey)
             requests += update_work(w)
```

Comment:
Will this work? How can we modify an array in parallel? I think we'd need to do some sort of map/reduce thing? Can we do that with joblib?

Reply:
You're right not a good idea to modify the array in parallel. I changed it so instead of modifying array it return it. Then we handle the modifying after the loops are done.